data/src/main/java/org/apache/iceberg/data/TableMigrationUtil.java

```diff
@@ -129,6 +140,9 @@ private TableMigrationUtil() {
             try {
               ParquetMetadata metadata = ParquetFileReader.readFooter(conf, stat);
               metrics = ParquetUtil.footerMetrics(metadata, Stream.empty(), metricsSpec, mapping);
+              if (!canImportSchema(ParquetSchemaUtil.convert(metadata.getFileMetaData().getSchema()), schema)) {
+                throw new ValidationException("Mismatch in table and imported file schema");
```

Comment:
Non-blocking question: Should we consider throwing the `ValidationException` from canImportSchema, so that the non-matching fields will be most visible in the initial exception message?

I can picture the support requests where the answer is "scroll up in the logs" just flooding in 🙈. Might be wise to cut that off as much as possible from the start.

Reply:
Also, each of the format specific listing dont have to throw the exception. Have added it